### ISSUE TITLE
bittrex-feed subscribe()

### ISSUE THREAD

**Issue by themagicmountain** [coinbase/gdax-tt#78]
should be asynchronous, and return a Promise<boolean> instead of boolean, because the two 
this.client.call()s in the function are asynchronous, and a successful subscription is dependent on their results

**Comment by themagicmountain** [comment]
sorry, maybe actually something like this,but i am not sure how to flatten the Promise.all array yet...

```typescript

subscribe(products: string[]): Promise<any> {
        if (!this.connection) {
            return new Promise((resolve, reject) => resolve(false));
        }
        return Promise.all(products.map((product: string) => {
            return Promise.all([
                new Promise((resolve, reject) => {
                    this.client.call('CoreHub', 'SubscribeToExchangeDeltas', product).done((err: Error, result: boolean) => {
                        if (err) {
                            return reject(err);
                        }

                        if (result === true) {
                            this.log('info', `Subscribed to ${product} on ${this.owner}`);
                            resolve(true);
                        }
                        resolve(false);
                    })
                }),
                new Promise((resolve, reject) => {
                    this.client.call('CoreHub', 'queryExchangeState', product).done((err: Error, data: any) => {
                        if (err) {
                            return reject(err);
                        }
                        if (!data) {
                            this.log("error", `failed to subscribe to ${product} on ${this.owner}`);
                            return resolve(false);
                        }
                        const snapshot: SnapshotMessage = this.processSnapshot(product, data);
                        this.push(snapshot);
                        resolve(true);
                    });
                })
            ])
        }));
    }
```

**Comment by CjS77** [comment]
Fixed in #102
